src/figuresClasses.ts

```diff
@@ -1,19 +1,110 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+abstract class BaseFigure implements Figure {
+  abstract shape: Shape;
 
+  abstract color: Color;
+
+  abstract getArea(): number;
 }
 
-export class Circle {
+export class Triangle extends BaseFigure {
+  shape = Shape.Triangle;
```

Comment:
I think the shape can be protected from changes.
```suggestion
  readonly shape = Shape.Triangle;
```